open-cluster-management/backlog#6858 Rem observ

cadawson opened the pull request.

cadawson committed:
open-cluster-management/backlog#6858 Rem observ

dockerymick reviewed (approved):
Changes look good Chris, thanks for the update. Quick question, after the MultiClusterObservability is removed, are users able to reinstall it later? I will ask the observability team, just wanted you to know what i curious about

dislbenn commented on install/uninstall.adoc:
```diff
@@ -48,16 +54,27 @@ oc get mch -o yaml
+. Remove the MultiClusterObservability operator, if installed.
+.. If the MultiClusterObservability operator is installed, select the tab for _MultiClusterObservability_.
+.. Select the _Options_ menu for the MultiClusterObservability operator. 
+.. Select *Delete MulticClusterObservability*. 
```
I believe this should be **MultiClusterObservability**. There's an extra c.

dockerymick commented on install/uninstall.adoc:
```diff
@@ -48,16 +54,27 @@ oc get mch -o yaml
+----
+
 [#deleting-both-components-by-using-the-console]
 == Deleting both components by using the console
```
When you say "both" are you referring to the CRD and operator?

cadawson replied:
Removed both, as it doesn't mean the same thing anymore. That is what it meant, but I think it becomes confusing now.

cadawson committed:
open-cluster-management/backlog#6858 rev cmts

cadawson committed:
open-cluster-management/backlog#6858 Add script

cadawson commented:
@joeg-pro and @nathanweatherly : This PR is intended to resolve Issue #6858 (adding instructions for removing the Observability operator, if it is there) and add the installation clean-up script to the main uninstallation instructions. Let me know if you see any issues to be addressed. Thanks!

joeg-pro commented on install/uninstall.adoc:
```diff
@@ -28,6 +28,12 @@ oc project <namespace>
+. If you have installed the MultiClusterObservability custom resource, enter the following command to remove it:
++
+----
+oc delete multiclusterobservability --all
```
In some commands (like deleting the CSV) the doc includes `-n open-cluster-management` to force/indicate the target namespace.  I think we should be consistent and include that on all of these commands.

joeg-pro commented on install/uninstall.adoc:
```diff
@@ -28,6 +28,12 @@ oc project <namespace>
+. If you have installed the MultiClusterObservability custom resource, enter the following command to remove it:
++
+----
+oc delete multiclusterobservability --all
```
n some commands (like deleting the CSV) the doc includes -n open-cluster-management to force/indicate the target namespace. I think we should be consistent and include that on all of these commands.

cadawson replied:
Thanks @joeg-pro . I added the namespaces. I think we are ok now.

nathanweatherly replied:
@joeg-pro I think technically it should be `-n <namespace>` with the note

```
Replace `<namespace>` in the script with the name of the namespace where {product-title-short} was installed.
```

joeg-pro commented on install/uninstall.adoc:
```diff
@@ -39,6 +45,35 @@ It might take up to 20 minutes to complete the uninstall process. You can view t
 ----
 oc get mch -o yaml
```
Add '-n open-cluster-management` here too.

nathanweatherly replied:
Same as above-- shoudl it be generic? `-n <namespace>`

joeg-pro commented on install/uninstall.adoc:
```diff
@@ -48,18 +83,30 @@ oc get mch -o yaml
 oc delete subs --all
```
Add '-n open-cluster-management` here too.

nathanweatherly replied:
Same

joeg-pro commented on install/uninstall.adoc:
```diff
@@ -48,18 +83,30 @@ oc get mch -o yaml
+. Enter the following command to delete the ClusterServiceVersion:
++
+----
+oc delete clusterserviceversion -n open-cluster-management
```
I think this should have `--all` added to it at end.  (Not sure if it works as is.)

joeg-pro reviewed (changes requested):
Just some minor changes to command examples.

cadawson committed:
open-cluster-management/backlog#6858 Add ns

joeg-pro commented on install/uninstall.adoc:
```diff
@@ -28,38 +28,85 @@ oc project <namespace>
+. If you have installed the MultiClusterObservability custom resource, enter the following command to remove it:
++
+----
+oc delete multiclusterobservability --all -n open-cluster-management
```
In talking about this, Nathan pointed out something that I missed in making my first comments about adding `-n open-cluster-management`:  that fact that the first step of these instructions says to use `oc project` to switch the current namespace (which might typically be called open-cluster-management) means that its unnecessary to add the `-n` option on many of these commands.

I agree with Nathan's point.  Given step 1, the `oc` command examples shown for steps 2, 3, 5 and 6 don't need to have `-n open-cluster-management` on them.  Moreover, in light of us explicitly telling the customer to do step 1 then it seems silly to tell them also to put the `-n` on these commands.

So, sorry for the run-arround on this, but lets remove the `-n open-cluster-management` from the commands for steps 2, 3 5 and 6.  (Do not remove any `-n` options found in the "script" that is spelled out in step 4.)

cadawson replied:
No problem. It is easy enough to fix. I was wondering about that, but there was also another place in the docs where someone asked to have that added, so I didn't worry about it. I removed them, and we should be ready to go. @joeg-pro @nathanweatherly

joeg-pro reviewed (changes requested):
See single comment.  Sorry for reversing my recommendation re -n optoins.

cadawson committed:
open-cluster-management/backlog#6858 Remove ns

nathanweatherly commented on install/uninstall.adoc:
```diff
@@ -48,18 +83,30 @@ oc get mch -o yaml
+. Enter the following command to delete the ClusterServiceVersion:
++
+----
+oc delete clusterserviceversion -all
```
--all (two dashes)

nathanweatherly commented on install/uninstall.adoc:
```diff
@@ -48,18 +83,30 @@ oc get mch -o yaml
-. Select the _Options_ menu for the MultiClusterHub operator.
-. Select *Delete MultiClusterHub*.
+. In the {ocp-short} console navigation, select *Operators* > *Installed Operators* > *Advanced Cluster Manager for Kubernetes*.
+. Remove the MultiClusterObservability operator, if installed.
```
"Remove the MultiClusterObservability Custom Resource, if created/installed"

nathanweatherly commented on install/uninstall.adoc:
```diff
@@ -48,18 +83,30 @@ oc get mch -o yaml
+.. If the MultiClusterObservability operator is installed, select the tab for _MultiClusterObservability_.
+.. Select the _Options_ menu for the MultiClusterObservability operator. 
+.. Select *Delete MultiClusterObservability*. 
+. Remove the MultiClusterHub operator.
```
"Remove the MultiClusterHub Custom Resource"

nathanweatherly commented on install/uninstall.adoc:
```diff
@@ -48,18 +83,30 @@ oc get mch -o yaml
+. In the {ocp-short} console navigation, select *Operators* > *Installed Operators* > *Advanced Cluster Manager for Kubernetes*.
+. Remove the MultiClusterObservability operator, if installed.
+.. If the MultiClusterObservability operator is installed, select the tab for _MultiClusterObservability_.
+.. Select the _Options_ menu for the MultiClusterObservability operator. 
```
"Select the _Options_ menu for the MultiClusterObservability." (I don't think work "operator" included)

nathanweatherly commented on install/uninstall.adoc:
```diff
@@ -48,18 +83,30 @@ oc get mch -o yaml
+.. Select the _Options_ menu for the MultiClusterObservability operator. 
+.. Select *Delete MultiClusterObservability*. 
+. Remove the MultiClusterHub operator.
+.. Select the tab for _Multiclusterhub operator_.
```
"... tab for _Multiclusterhub_"

nathanweatherly commented on install/uninstall.adoc:
```diff
@@ -48,18 +83,30 @@ oc get mch -o yaml
+.. Select *Delete MultiClusterObservability*. 
+. Remove the MultiClusterHub operator.
+.. Select the tab for _Multiclusterhub operator_.
+.. Select the _Options_ menu for the MultiClusterHub operator.
```
s/operator/custom resource

nathanweatherly commented on install/uninstall.adoc:
```diff
@@ -48,18 +83,30 @@ oc get mch -o yaml
-. Select *Delete MultiClusterHub*.
+. In the {ocp-short} console navigation, select *Operators* > *Installed Operators* > *Advanced Cluster Manager for Kubernetes*.
+. Remove the MultiClusterObservability operator, if installed.
+.. If the MultiClusterObservability operator is installed, select the tab for _MultiClusterObservability_.
```
If the MulticlusterObservability Custom Resource is installed, ...

cadawson committed:
open-cluster-management/backlog#6858 install rev

cadawson committed:
open-cluster-management/backlog#6858 move uninst

cadawson committed:
open-cluster-management/backlog#6858 Rem note

cadawson committed:
open-cluster-management/backlog#6858 rem disable

cadawson commented:
Ok. I didn't like my option to create a new file. Here is what I did:
1. I moved the steps that were in the observability section to the uninstalling file.
2. I removed the important note about removing the observability cr, since it is now covered inline in the uninstallation steps. 
3. I left the section title in the observability section so people who are only trying to disable observability can find something there. In the section, I link to the procedure in the uninstallation topic that contains the steps. 
I think this might be a good option. Let me know what you think. Thanks! @bjoydeep @nathanweatherly

bjoydeep commented on install/uninstall.adoc:
```diff
@@ -28,6 +26,20 @@ oc project <namespace>
+When you delete the resource, the pods in the `open-cluster-management-observability` namespace on {product-title-short} hub cluster, and the pods in `open-cluster-management-addon-observability` namespace on all managed clusters are removed. 
++
+*Important*: Your object storage is not affected after you remove the observability service.
+
```
@cadawson I would swap steps 1 and 2...
you log into your cluster first
run `oc delete mco observability`
then - change to project namespace perhaps `oc project open-cluster-management`

bjoydeep commented on observability/observability_install.adoc:
```diff
@@ -138,22 +138,11 @@ All the pods in `open-cluster-management-observability` namespace for Thanos, Gr
 
-Disable the observability service by removing the `observability` resource that you created. Complete the following steps:
+To disable the observability service, uninstall the `observability` resource. See steps 1 and 2 of link:../install/#removing-a-multiclusterhub-instance-by-using-commands[Removing a MultiClusterHub instance by using commands].
 
```
therefore would say `See Step 1` Only and not step 2

cadawson committed:
open-cluster-management/backlog#6858 Switch steps

cadawson committed:
open-cluster-management/backlog#6858 ref step 1

cadawson commented:
FYI: @dockerymick After a couple of iterations, we moved the removal of the observability cr to the uninstallation topic. For those who look for it in the installing observability topic, we left a link to the uninstallation topic. This change was made in the 2.1 and 2.2 versions.
